code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaData.java

```diff
@@ -39,6 +40,8 @@ public class ContentCardSchemaData implements SchemaData {
                             schemaData.optString(
                                     MessagingConstants.ConsequenceDetailDataKeys.CONTENT_TYPE));
             if (contentType.equals(ContentType.APPLICATION_JSON)) {
+                this.contentJsonString =
+                        schemaData.getString(MessagingConstants.ConsequenceDetailDataKeys.CONTENT);
                 this.content =
                         JSONUtils.toMap(
```

Comment:
Since we're already converting to map here, wouldn't it be more efficient to use the map rather than convert it to json again and read the keys? Isn't read from map faster than json?

Reply:
Another advantage of using map is you can directly use `DataReader` provided by Core that already does a lot of null checks and error handling